pkg/core/consensus/user/keys.go

```diff
@@ -21,7 +21,12 @@ type Keys struct {
 // NewRandKeys will generate and return new bls and ed25519
 // keys to be used in consensus
 func NewRandKeys() (Keys, error) {
-	return NewKeysFromReader(nil)
+	keys, err := NewKeysFromReader(nil)
+	if err == io.EOF {
+		return NewRandKeys()
```

Comment:
Is there a reason for using recursion instead of a simple loop here?

Reply:
It's more concise, I think. Other than that, no reason.

Reply:
In general I would avoid recursion. In this case though, recursion helps the conciseness and I find it more elegant than a 'for' with a nested if. At the end of the day the recursion is a rare circumstance while a 'for' would suggest that an EOF is expected to happen less infrequently

Reply:
My point was this would grow the stack indefinitely. If a `for` was used the compiler would choose to simply inline `NewRandKeys`

Reply:
Only if the recursion actually happens, which is a rare occurrence. However, point taken. Feel free to revert to a for loop